Code/Framework/AzCore/AzCore/Math/MathUtils.h

```diff
@@ -461,6 +461,36 @@ namespace AZ
         return (azisfinite(x) != 0);
     }
 
+    //! Returns the value divided by alignment, where the result is rounded up if the remainder is non-zero.
+    //! Example: alignment: 4
+    //! Value:  0 1 2 3 4 5 6 7 8
+    //! Result: 0 1 1 1 1 2 2 2 2
+    constexpr uint32_t DivideAndRoundUp(uint32_t value, uint32_t alignment)
+    {
+        AZ_Assert(alignment != 0, "0 is an invalid multiple to round to.");
+        return ((value + alignment - 1) / alignment);
```

Comment:
question: Is it worth testing or asserting/warning about an input that is equal to INT_MAX (just thinking of potentially hairy edge cases people might inadvertently run into).

Reply:
> question: Is it worth testing or asserting/warning about an input that is equal to INT_MAX (just thinking of potentially hairy edge cases people might inadvertently run into).

Hm, more generally, you could run into issues if value + alignment > INT_MAX